endpoints/apiconfig.go

```diff
@@ -777,7 +796,16 @@ func validateField(v reflect.Value, t reflect.StructField) error {
 
 	isZero := v.Interface() == reflect.Zero(v.Type()).Interface()
 	if isZero && tag.required {
-		return fmt.Errorf("missing field %v", t.Name)
+		return fmt.Errorf("missing field %v in %v", t.Name, parent.Name())
+	}
+
+	if v.Kind() == reflect.Struct {
+		if v.CanAddr() && hasExportedFields(v) {
```

Comment:
Haven't seen ti, but seemed like a safe way to do it, but you are right about returning an error. 
Ok, so you decide, either I remove the `CanAddr` check or add an error message.